fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/SemiReorderRule.java

```diff
@@ -142,10 +144,12 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         // build new semi join projection
         if (semiExpression.isEmpty()) {
             newSemiJoin = new LogicalJoinOperator.Builder().withOperator(topJoin)
+                    .setLimit(Operator.DEFAULT_LIMIT)
                     .setProjection(new Projection(projectMap)).build();
         } else {
             semiExpression.putAll(projectMap);
             newSemiJoin = new LogicalJoinOperator.Builder().withOperator(topJoin)
+                    .setLimit(Operator.DEFAULT_LIMIT)
```

Comment:
why limit property cannot derive from the `topJoin` and must be set to -1?

Reply:
Open you mind, think the sql 
```
select * from (select A join B) join C limit 1

select * from (select B join C limit 1) join A
```